packages/server-wallet/src/protocols/__test__/close-channel.test.ts

```diff
@@ -1,70 +1,151 @@
+import {testKnex as knex} from '../../../jest/knex-setup-teardown';
+import {MockChainService} from '../../chain-service';
+import {defaultTestConfig} from '../../config';
+import {createLogger} from '../../logger';
 import {ChainServiceRequest, requestTimeout} from '../../models/chain-service-request';
-import {channel} from '../../models/__test__/fixtures/channel';
+import {DBCloseChannelObjective} from '../../models/objective';
+import {Store} from '../../wallet/store';
+import {WalletResponse} from '../../wallet/wallet-response';
+import {TestChannel} from '../../wallet/__test__/fixtures/test-channel';
+import {ChannelCloser} from '../close-channel';
 
-import {applicationProtocolState} from './fixtures/protocol-state';
+const FINAL = 10; // this will be A's state to sign
+const testChan = TestChannel.create({aBal: 5, bBal: 5, finalFrom: FINAL});
 
-const outcome = simpleEthAllocation([{amount: BN.from(5), destination: alice().destination}]);
+let store: Store;
 
-const participants = [alice(), bob()];
-const postFundState = {outcome, turnNum: 3, participants};
-const closingState = {outcome, turnNum: 4, isFinal: true, participants};
+beforeEach(async () => {
+  store = new Store(
+    knex,
+    defaultTestConfig().metricsConfiguration.timingMetrics,
+    defaultTestConfig().skipEvmValidation,
+    '0'
+  );
+  await store.dbAdmin().truncateDB();
+});
 
-const runningState = {outcome, turnNum: 7, participants};
-const closingState2 = {outcome, turnNum: 8, isFinal: true, participants};
+afterEach(async () => await store.dbAdmin().truncateDB());
 
-const signState = (state: Partial<State>): Partial<SignState> => ({type: 'SignState', ...state});
-const withdrawAction = {
-  type: 'Withdraw',
-  channelId: channel().channelId,
-};
-const validCSR = ChainServiceRequest.fromJson({
-  channelId: 0,
-  request: 'withdraw',
-  timestamp: new Date(),
+describe(`closing phase`, () => {
+  it(`waits, if it isn't my turn`, async () => {
+    const objective = await setup(testChan, {participant: 0, statesHeld: [7, 8]});
+    await crankAndAssert(objective, {statesToSign: []});
+  });
+  it(`creates and signs the closing state, if it is my turn`, async () => {
+    const objective = await setup(testChan, {participant: 0, statesHeld: [8, 9]});
+    await crankAndAssert(objective, {statesToSign: [FINAL]});
+  });
+  it(`double signs the closing state and withdraws, if my opponent has already signed a closing state`, async () => {
+    const objective = await setup(testChan, {participant: 1, statesHeld: [9, FINAL]});
+    await crankAndAssert(objective, {statesToSign: [FINAL + 1], withdraws: true});
+  });
 });
 
-const staleCSR = ChainServiceRequest.fromJson({
-  channelId: 0,
-  request: 'withdraw',
-  timestamp: new Date(Date.now() - requestTimeout - 1),
-});
+describe(`defunding phase (when the channel is closed)`, () => {
+  describe(`direct funding`, () => {
+    it('submits the withdrawal transaction', async () => {
```

Comment:
The description should reflect the assertions better. 
```suggestion
    it('submits the withdrawal transaction, and submits once again when it becomes stale, but not a third time', async () => {
```
^ is this what you are actually testing? why do we have that behaviour?

Reply:
I'm not sure why this is the behaviour. All I've done here is replicate the test the @kerzhner added for open channel test.

Reply:
Good call on a testcase description that does not match the logic. At the moment, the wallet attempts each chain service request (for each channel) twice: https://github.com/statechannels/statechannels/pull/3001.

Happy to reconsider this heuristic!